dockernet/src/init_chain.sh

```diff
@@ -213,35 +229,34 @@ if [ "$CHAIN" == "STRIDE" ]; then
         relayer_acct="${STRIDE_RELAYER_ACCTS[i]}"
         relayer_mnemonic="${STRIDE_RELAYER_MNEMONICS[i]}"
 
-        echo "$relayer_mnemonic" | $MAIN_CMD keys add $relayer_acct --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
-        
-        relayer_address=$($MAIN_CMD keys show $relayer_acct --keyring-backend test -a)
-        $MAIN_CMD add-genesis-account ${relayer_address} ${VAL_TOKENS}${DENOM}
+        add_relayer_account "$relayer_acct" "$relayer_mnemonic"
     done
 else 
     # add a revenue account
     REV_ACCT_VAR=${CHAIN}_REV_ACCT
     REV_ACCT=${!REV_ACCT_VAR}
     echo $REV_MNEMONIC | $MAIN_CMD keys add $REV_ACCT --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
 
-    # add a relayer account
-    relayer_acct=$(GET_VAR_VALUE     RELAYER_${CHAIN}_ACCT)
-    relayer_mnemonic=$(GET_VAR_VALUE RELAYER_${CHAIN}_MNEMONIC)
+    # add a relayer account if the chain is a HOST_CHAIN
+    # if it's only an accessory chain, the account will be added after the network is started
+    is_host_chain=false
+    for host_chain in ${HOST_CHAINS[@]}; do
+        if [ "$CHAIN" == "$host_chain" ]; then 
+            relayer_acct=$(GET_VAR_VALUE     RELAYER_${CHAIN}_ACCT)
+            relayer_mnemonic=$(GET_VAR_VALUE RELAYER_${CHAIN}_MNEMONIC)
 
-    echo "$relayer_mnemonic" | $MAIN_CMD keys add $relayer_acct --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
-    relayer_address=$($MAIN_CMD keys show $relayer_acct --keyring-backend test -a | tr -cd '[:alnum:]._-')
-    $MAIN_CMD add-genesis-account ${relayer_address} ${VAL_TOKENS}${DENOM}
+            add_relayer_account "$relayer_acct" "$relayer_mnemonic"
+            break
+        fi
+    done
 
+    # gaia ICS and noble have a different config setup, so we handle them explicitly here
     if [ "$CHAIN" == "GAIA" ]; then 
-        echo "$RELAYER_GAIA_ICS_MNEMONIC" | $MAIN_CMD keys add $RELAYER_GAIA_ICS_ACCT --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
-        relayer_address=$($MAIN_CMD keys show $RELAYER_GAIA_ICS_ACCT --keyring-backend test -a | tr -cd '[:alnum:]._-')
-        $MAIN_CMD add-genesis-account ${relayer_address} ${VAL_TOKENS}${DENOM}
+        add_relayer_account "$RELAYER_GAIA_ICS_ACCT" "$RELAYER_GAIA_ICS_MNEMONIC"
     fi
-
-    if [ "$CHAIN" == "DYDX" ]; then
-        echo "$RELAYER_DYDX_NOBLE_MNEMONIC" | $MAIN_CMD keys add $RELAYER_DYDX_NOBLE_ACCT --recover --keyring-backend test >> $KEYS_LOGS 2>&1
-        relayer_address=$($MAIN_CMD keys show $RELAYER_DYDX_NOBLE_ACCT --keyring-backend test -a | tr -cd '[:alnum:]._-')
-        $MAIN_CMD add-genesis-account ${relayer_address} ${VAL_TOKENS}${DENOM}
+    if [ "$CHAIN" == "NOBLE" ]; then 
```

Comment:
I know these are the relayer addresses not the init on the ICA addresses, but it reminded me about the reward zone ICA.   

If noble has the most up to date pfm module, then we don't need to create the reward zone unwinding ICA at all, if the address is input as "pfm" the most recent pfm module is supposed to be able to handle it if a non-legal address is given.  But we should test this in different versions of integration scripts by changing the TradeRoute data we initialize, one with valid address on the reward zone and one where the address is simply "pfm"

Reply:
agreed, good call!